birdbox/birdbox/settings/base.py

```diff
@@ -418,14 +418,16 @@ def path_from_root(*args):
 
 # Security settings (see `manage.py check --deploy`)
 
-DISABLE_SSL = config("DISABLE_SSL", default=str(DEBUG), parser=bool)  # so default: "False"
 SECURE_SSL_REDIRECT = config(
     "SECURE_SSL_REDIRECT",
-    default=str(not DISABLE_SSL),
+    default="False",  # Deliberately off by default - we don't need to upgrade at the app level
     parser=bool,
-)  # so default: "True"
-if config("USE_SECURE_PROXY_HEADER", default=str(SECURE_SSL_REDIRECT), parser=bool):
+)
+if config("USE_SECURE_PROXY_HEADER", default="True", parser=bool):
```

Comment:
We shouldn't need this anymore either, right? Or it should at least default to `False`?

Reply:
Ah, I assumed we did need it, cos the CDN/LB infra was terminating HTTPS and forwarding on. 

My thought process was more or less: Because we know it'll all be over HTTPS maybe we don't need it, but if it's not used then [the docs](https://docs.djangoproject.com/en/3.2/ref/settings/#secure-proxy-ssl-header) ready like  `HttpRequest.is_secure()` may return `False`, which could break other things (eg CSRF for the Wagtail side).

You could be totally right, though! I'll think on it a bit more

Reply:
due to it already being HTTPS, as guaranteed by the CDN, someone sending a request with `x-forwarded-proto: http` could poison the cache because that would cause Django to return a redirect to `https` which would then loop because the redirect response would be cached.

Reply:
unless GCP CDN is sending an accurate `x-forwarded-proto` header. it seems cloudfront is not.

Reply:
I think it should set it

Reply:
hmm so https://cloud.google.com/load-balancing/docs/https#target-proxies says `x-forwarded-proto` is set by different load balancer options but it's hard to determine whether Cloud CDN nixes that. @bkochendorfer Are we using the classic LB or the Global external Application Load Balancer please?

Reply:
GKE Ingress which is a ` Classic Application Load Balancer `

Reply:
I've set this to default to False, but still be tweakable via env vars. Will merge it and see how it is on dev

Reply:
Ugh so disabling the SSL_REDIRECT upgrade breaks the redirect URI generated by for django-oicd, breaking SSO. I've just reverted it on `main` to bring dev back in line. I think I'll try the wsgi trick Pmac mentions above in case that gives us the best of both worlds: no HTTP->S upgrade and a signal that we're on HTTPS (because effectively we will be)